### PR TITLE
Update API tests using factories

### PR DESCRIPTION
**Pull request by aaarendt**
Issue #158 

Updates test_layer_measurements.py following the new test infrastructure using factories. 
Adds a new factory with real layer density data.

Showing progress so far to ensure I'm on the right track.

### CODE REVIEW

File: tests/api/test_layer_measurements.py [hunk 1]
```diff
@@ -1,111 +1,105 @@
...
+        assert result == [
+            record.site.date
+            for record in self.db_data
+        ]
 
-    def test_all_observers(self, clz):
-        result = clz().all_observers
-        assert result == ['TEST']
+@pytest.fixture
+def layer_data(layer_density_factory, db_session):
```
Comment (jomey) [inline tests/api/test_layer_measurements.py:58]: Suggest of not re-using the same method name within a file and pick something unique (e.g. layer_data_with_density)

**Review by jomey — APPROVED** [review]
Looks very good! :1st_place_medal:

File: tests/api/test_layer_measurements.py (hunk 1 above)
Comment (micah-prime) [reply tests/api/test_layer_measurements.py:58]: Yeah I think that is a good call

File: tests/api/test_layer_measurements.py [hunk 2]
```diff
@@ -1,111 +1,105 @@
...
 
-    def test_from_area(self, clz):
-        df = gpd.GeoDataFrame(
-            geometry=gpd.points_from_xy(
-                [743766.4794971556], [4321444.154620216], crs="epsg:26912"
-            ).buffer(1000.0)
-        ).set_crs("epsg:26912")
-        result = clz.from_area(
-            type="density",
-            shp=df.iloc[0].geometry,
```
Comment (micah-prime) [inline tests/api/test_layer_measurements.py:100]: I'm hesitant to lose these from_area tests, I think it is still important functionality to have safeguards around

**Comment by micah-prime** [comment]
This is looking good!

File: tests/api/test_layer_measurements.py (hunk 2 above)
Comment (jomey) [reply tests/api/test_layer_measurements.py:100]: Nice catch. I think this test can be transferred over to the new structure.

Comment (aaarendt) [reply tests/api/test_layer_measurements.py:100]: Agreed, I brought back those tests.

**Review by micah-prime — APPROVED** [review]

**Review by jomey — APPROVED** [review]

**Pull request merged**